src/NexusMods.App.UI/Pages/ModLibrary/FileOrigins/FileOriginsPageViewModel.cs

```diff
@@ -41,35 +43,39 @@ public class FileOriginsPageViewModel : APageViewModel<IFileOriginsPageViewModel
         _conn = conn;
         _archiveInstaller = archiveInstaller;
         _dlAnalysisRepo = downloadAnalysisRepository;
-        
+
         TabTitle = Language.FileOriginsPageTitle;
         TabIcon = IconValues.ModLibrary;
-        
+
         LoadoutId = loadoutId;
-        
+
         var loadout = _conn.Db.Get<Loadout.Model>(LoadoutId.Value);
         var game = loadout.Installation.Game;
-        
-        var downloadAnalyses = _dlAnalysisRepo.Observable;
-        
-        var entriesObservable = downloadAnalyses.ToObservableChangeSet()
-            .Filter(downloadAnalysis => (!downloadAnalysis.TryGet(DownloaderState.GameDomain, out var domain) 
-                                         || domain == game.Domain)
-                                        && !downloadAnalysis.Contains(StreamBasedFileOriginMetadata.StreamBasedOrigin))
-            .OnUI()
-            .Transform(fileOrigin => (IFileOriginEntryViewModel)new FileOriginEntryViewModel(
-                    _conn,
-                    _archiveInstaller,
-                    LoadoutId,
-                    fileOrigin
+
+        var entriesObservable = downloadAnalysisRepository.Observable
+                .ToObservableChangeSet()
+                .Filter(model => FilterDownloadAnalysisModel(model, game.Domain))
+                .OnUI()
+                .Transform(fileOrigin => (IFileOriginEntryViewModel)new FileOriginEntryViewModel(
+                        _conn,
+                        _archiveInstaller,
+                        LoadoutId,
+                        fileOrigin
+                    )
                 )
-            )
-            .Bind(out _fileOrigins);
-        
+                .Bind(out _fileOrigins);
+
         this.WhenActivated(d =>
         {
-            entriesObservable.Subscribe()
-                .DisposeWith(d);
+            entriesObservable.SubscribeWithErrorLogging().DisposeWith(d);
         });
     }
+
+    public static bool FilterDownloadAnalysisModel(DownloadAnalysis.Model model, GameDomain currentGameDomain)
+    {
+        if (!model.TryGet(DownloaderState.GameDomain, out var domain)) return false;
```

Comment:
This should not return false
If we don't have a GameDomain on the DownloadState it is likely because it was installed from archive on disk, but we want to show those, not filter them out.

You can move the StreamBasedFileOrigin check first, then return true if has no game, then false if game doesn't match

Reply:
Why do archives from disk not have a game domain? The button to do so is in the Loadout Grid, so we have access to the game domain.

Reply:
The problem with that is that you are assuming that the user will add a mod for this game, which isn't necessarily true.
So someone on SSE could add a mod for Legendary edition and we would auto label it as SSE mod. Leaving it unknown is preferrable in that case. 
I guess we will have to figure something out when we add support for another game.

Reply:
Updated the code to add `GameDomain` to the manually added archive.

Reply:
We don't support SSE or LE yet. For Stardew Valley, modders would expect their Stardew Valley archives to appear for Stardew Valley.

Reply:
You can add a random archive and it would show up as a StardewValley mod, which doesn't make too much sense.
We probably need a user facing value and an internal value.

Reply:
I don't see why we'd want to be so complex about this. If the user is on a Stardew Valley loadout and manually adds an archive, then we should accept that as a Stardew Valley archive.

Reply:
The file would show up even if installation failed because it wasn't actually a StardewMod. We might want to ask @captainsandypants

Reply:
Mod installation isn't relevant to this, tho. Adding a mod manually is an escape hatch in the first place. Again, this shouldn't be complex.

Reply:
I agree that if the user adds a mod manually to Stardew we should register is as Stardew, although where would the game domain show up in the UI? And do we need to consider multiple game store version of Stardew being managed for this?

Reply:
> although where would the game domain show up in the UI?

We don't have a column that shows the game domain, but the code above filters by game domain because that's the only way to check if the download is for the current game.

> And do we need to consider multiple game store version of Stardew being managed for this?

No. The "game domain" is the identifier used by Nexus Mods `stardewvalley` and applies to all versions of the game. It's our way of uniquely identifying a game across installations.